fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -173,41 +170,55 @@ private void validate(Map<String, String> properties) throws DdlException {
         }
     }
 
-    private Set<PrimitiveType> getValidColumnType(Type.TypeID icebergType) {
+    private boolean validateColumnType(Type icebergType, com.starrocks.catalog.Type type) {
         if (icebergType == null) {
-            return Sets.newHashSet();
+            return false;
         }
 
+        if (icebergType.isListType()) {
+            return validateColumnType(icebergType.asListType().elementType(), ((ArrayType) type).getItemType());
+        }
+
+        if (!icebergType.isPrimitiveType()) {
+            return false;
+        }
+        PrimitiveType primitiveType = type.getPrimitiveType();
         // for type with length, like char(10), we only check the type and ignore the length
         // TODO: fixed and binary should be considered as binary
-        switch (icebergType) {
+        switch (icebergType.typeId()) {
             case BOOLEAN:
-                return Sets.newHashSet(PrimitiveType.BOOLEAN);
+                return primitiveType == PrimitiveType.BOOLEAN;
             case INTEGER:
-                return Sets.newHashSet(PrimitiveType.INT, PrimitiveType.TINYINT, PrimitiveType.SMALLINT);
+                return primitiveType == PrimitiveType.INT ||
+                        primitiveType == PrimitiveType.TINYINT ||
+                        primitiveType == PrimitiveType.SMALLINT;
             case TIME:
             case TIMESTAMP:
             case LONG:
-                return Sets.newHashSet(PrimitiveType.BIGINT);
+                return primitiveType == PrimitiveType.BIGINT;
             case FLOAT:
-                return Sets.newHashSet(PrimitiveType.FLOAT);
+                return primitiveType == PrimitiveType.FLOAT;
             case DOUBLE:
-                return Sets.newHashSet(PrimitiveType.DOUBLE);
+                return primitiveType == PrimitiveType.DOUBLE;
             case DATE:
-                return Sets.newHashSet(PrimitiveType.DATE, PrimitiveType.DATETIME);
+                return primitiveType == PrimitiveType.DATE ||
+                        primitiveType == PrimitiveType.DATETIME;
             case STRING:
             case UUID:
-                return Sets.newHashSet(PrimitiveType.VARCHAR, PrimitiveType.CHAR);
+                return primitiveType == PrimitiveType.VARCHAR ||
+                        primitiveType == PrimitiveType.CHAR;
             case DECIMAL:
-                return Sets.newHashSet(PrimitiveType.DECIMALV2, PrimitiveType.DECIMAL32,
-                        PrimitiveType.DECIMAL64, PrimitiveType.DECIMAL128);
+                return primitiveType == PrimitiveType.DECIMALV2 ||
+                        primitiveType == PrimitiveType.DECIMAL32 ||
+                        primitiveType == PrimitiveType.DECIMAL64 ||
+                        primitiveType == PrimitiveType.DECIMAL128;
             case FIXED:
             case BINARY:
             case STRUCT:
             case LIST:
```

Comment:
this case conflict with icebergType.isListType() checker previous?

Reply:
Not really, this case would never reach by define